### PR TITLE
feat: chaos send (partially deprecate puppet actor)

### LINKED ISSUE #61: consolidate the Puppet and the Chaos actor

### PR DESCRIPTION
**Pull request by alanshaw**
Adds `Send` functionality from the puppet actor removing dependence on it from this repo.

Fixes #61.

### CODE REVIEW

File: gen/builders/messages_typed.go [hunk 1]
```diff
@@ -363,17 +363,12 @@ func InitExec(params *init_.ExecParams) TypedCall {
 }
 
 // ----------------------------------------------------------------------------
-// | PUPPET
+// | CHAOS
 // ----------------------------------------------------------------------------
 
-func PuppetConstructor(params *adt.EmptyValue) TypedCall {
+// ChaosSend is a message to a chaos actor that sends a messgae to another actor.
```
Comment (raulk) [inline gen/builders/messages_typed.go:369]:
```suggestion
// ChaosSend is a message to a chaos actor that sends a message to another actor.
```

File: chaos/actor.go [hunk 2]
```diff
@@ -49,11 +51,57 @@ func (a Actor) Exports() []interface{} {
...
+func handleSendReturn(ret runtime.SendReturn) (runtime.CBORBytes, error) {
+	if ret == nil {
+		return nil, nil // nothing was returned
+	}
+	var out runtime.CBORBytes
+	if err := ret.Into(&out); err != nil {
+		return nil, err
+	}
+	return out, nil
+}
```
Comment (raulk) [inline chaos/actor.go:103]: Can we inline this, since it's called only in one place?

**Review by raulk — COMMENTED** [review]
Yay! This looks great. I fixed the description to say it "fixes #61", so that that issue is closed.

I guess you marked this as a WIP PR because we need to remove the puppet actor instantiation from the driver in Lotus? What other work is there left to do?

We can probably rejig the prose a little in the README to remove the bullet list since we have only one test harness actor now.

**Comment by alanshaw** [comment]
@raulk I just pulled in the parts of puppet that were required here in `test-vectors`. There is additional functionality in puppet:

1. It allows construction of additional puppets with an initial state (e.g. balance)
2. It has extra methods `SendMarshalCBORFailure`, `ReturnMarshalCBORFailure` and `RuntimeTransactionMarshalCBORFailure` https://github.com/filecoin-project/specs-actors/blob/master/actors/puppet/puppet.go#L72-L105

The `chain-validation` tests do not use this functionality (neither do we). Might be best to re-add this functionality if/when we need it?

**Comment by alanshaw** [comment]
[I did some spelunking](https://github.com/filecoin-project/specs-actors/pull/399#issuecomment-638331658) and found that the additional puppet methods will be needed for https://github.com/filecoin-project/test-vectors/issues/57

File: README.md [hunk 3]
```diff
@@ -255,25 +255,20 @@ $ make regen
...
- * **Puppet Actor (address `t98`):** former testing harness that includes
-   methods for invalid sending messages and returning invalid values. It's in
-   the process of being deprecated and consolidated into the **Chaos actor**
-   (track [#61](https://github.com/filecoin-project/test-vectors/issues/61).
-     * Test vectors requiring the **Puppet actor** carry the `puppet_actor:true`
-       selector. 
+those vectors rely on a special test harness actor that sits "on the inside"
+and triggers those situations when specific messages are sent to it.
+
+The **Chaos Actor (address `t97`)** exercises behaviours that should be
```
Comment (raulk) [inline README.md:261]: WDYT about replacing this with `t98` (replacing the puppet actor, now that it's out of the picture)?

We also need to update the driver on Lotus side. Could you tackle that?

**Review by raulk — APPROVED** [review]
Let's tackle the remaining work in a follow-up PR:

- [ ] make the chaos actor `t98` (all vectors will be affected).
- [ ] PR to decommission the puppet actor from the driver (Lotus side).

**Pull request merged**